numba/annotations/type_annotations.py

```diff
@@ -98,7 +98,7 @@ def prepare_annotations(self):
                         atype = 'XXX Lifted Loop XXX'
                         found_lifted_loop = True
                     else:
-                        atype = self.typemap[inst.target.name]
+                        atype = self.typemap.get(inst.target.name, "<missing>")
```

Comment:
(Self note) check why.

Reply:
This is a problem caused by parfor lowering.  Likely that an Assign node is added without updating the typemap.

Reply:
Need to open issue to note this.

Reply:
opened #4212

Reply:
@sklam Any suggestions for minimal reproducer?  If you give me a little guidance in terms of how to reproduce then I'll take a look.

Reply:
If you undo this changes, the `parallel=True` functions in https://github.com/numba/numba/pull/4155/files#diff-8143656cacb63791a83889ff0bcee7d7R9 will fail due to missing variables in the `typemap`.